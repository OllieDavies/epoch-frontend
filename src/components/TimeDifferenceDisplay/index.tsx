import * as React from "react";
import { useState, useEffect } from "react";

interface ITimeDifferenceDisplayProps {
  startEpoch: number;
}

const TimeDifferenceDisplay: React.FC<ITimeDifferenceDisplayProps> = ({
  startEpoch,
}) => {
  const [seconds, setSeconds] = useState(0);

  useEffect(() => {
    function updateSeconds() {
      const seconds = Math.floor(new Date().getTime() - startEpoch * 1000);
      setSeconds(seconds);
    }

    const timer = setInterval(updateSeconds, 1000);
    return () => {
      clearInterval(timer);
    };
  });

  return <>{new Date(seconds).toISOString().substr(11, 8)}</>;
};

export default TimeDifferenceDisplay;
