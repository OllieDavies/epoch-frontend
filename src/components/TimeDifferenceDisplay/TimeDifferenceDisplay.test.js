import React from "react";
import TimeDifferenceDisplay from "./index";

describe("TimeDifferenceDisplay",()=> {
  test("matches snapshot",() =>{
    expect(<TimeDifferenceDisplay startEpoch={1000000} />).toMatchSnapshot()
  })
  test.todo("converts the number of seconds into the correct HH:mm:ss format")
  test.todo("updates every second")
})