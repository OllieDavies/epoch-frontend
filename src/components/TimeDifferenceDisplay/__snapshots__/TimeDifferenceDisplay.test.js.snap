// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeDifferenceDisplay matches snapshot 1`] = `
<TimeDifferenceDisplay
  startEpoch={1000000}
/>
`;
